pkg/util/steps/condition.go

```diff
@@ -66,9 +86,27 @@ func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
 		log.Warnf("step %s failed but has configured 'fail=%t'. Continuing. Error: %s", c, c.fail, err.Error())
 		return nil
 	}
+	if err == wait.ErrWaitTimeout {
+		return c.enrichConditionTimeoutError()
+	}
 	return err
 }
 
+func (c conditionStep) enrichConditionTimeoutError() error {
```

Comment:
this utility function is prime for a test. Please add some coverage.

Reply:
I have added test coverage for enrichConditionTimeoutError. Thanks